pandas/core/arrays/categorical.py

```diff
@@ -2205,7 +2205,10 @@ def max(self, *, skipna=True, **kwargs):
             pointer = self._codes.max()
         return self._wrap_reduction_result(None, pointer)
 
-    def mode(self, dropna=True):
+    def _mode(self, dropna: bool = True) -> Categorical:
+        return self.mode(dropna=dropna)
+
+    def mode(self, dropna: bool = True) -> Categorical:
```

Comment:
shouldn't we move the impl to _mode and deprecate mode here?

Reply:
yes.  next pass (getting the CI to green is a PITA these days)